apps/hash-frontend/src/pages/[shortname].page.tsx

```diff
@@ -0,0 +1,134 @@
+import { types } from "@local/hash-isomorphic-utils/ontology-types";
+import {
+  OrgProperties,
+  UserProperties,
+} from "@local/hash-isomorphic-utils/system-types/shared";
+import { Entity } from "@local/hash-subgraph";
+import { Container, Typography } from "@mui/material";
+import { NextParsedUrlQuery } from "next/dist/server/request-meta";
+import { useRouter } from "next/router";
+import { useMemo, useState } from "react";
+
+import { constructOrg, constructUser } from "../lib/user-and-org";
+import { getLayoutWithSidebar, NextPageWithLayout } from "../shared/layout";
+import { useUserOrOrg } from "../shared/use-user-or-org";
+import { EditUserProfileInfoModal } from "./[shortname].page/edit-user-profile-info-modal";
+import { ProfilePageContent } from "./[shortname].page/profile-page-content";
+import { ProfilePageHeader } from "./[shortname].page/profile-page-header";
+import { useAuthenticatedUser } from "./shared/auth-info-context";
+
+export const parseProfilePageUrlQueryParams = (
+  queryParams: NextParsedUrlQuery | undefined,
+) => {
+  const paramsShortname = queryParams?.shortname;
+
+  if (!paramsShortname || typeof paramsShortname !== "string") {
+    throw new Error("Could not parse `shortname` from query params.");
+  }
+
+  const profileShortname = paramsShortname.slice(1);
+
+  return { profileShortname };
+};
+
+export const leftColumnWidth = 150;
+
+const ProfilePage: NextPageWithLayout = () => {
+  const router = useRouter();
+  const { authenticatedUser } = useAuthenticatedUser();
+
+  const { profileShortname } = parseProfilePageUrlQueryParams(router.query);
+
+  const [displayEditUserProfileInfoModal, setDisplayEditUserProfileInfoModal] =
+    useState(false);
+
+  const { userOrOrg, userOrOrgSubgraph, loading, refetch } = useUserOrOrg({
+    shortname: profileShortname,
+    graphResolveDepths: {
+      // Required to retrieve avatars. Will need amending if we want an org's memberships (they are incoming links)
+      hasLeftEntity: { incoming: 1, outgoing: 0 },
+      hasRightEntity: { incoming: 0, outgoing: 1 },
```

Comment:
**Blocking:** The check for `isEditable` below will not work for orgs because we aren't fetching their memberships.

Two options:
1. Fetch the membership entity only, and check that the rightEntityId matches the user's entity id (I'd prefer this unless we need to show user details on the org page)
```suggestion
      hasLeftEntity: { incoming: 1, outgoing: 1 },
      hasRightEntity: { incoming: 1, outgoing: 1 },
```

2. Fetch the user as well
```suggestion
      hasLeftEntity: { incoming: 2, outgoing: 1 },
      hasRightEntity: { incoming: 1, outgoing: 2 },
```

Reply:
Thanks Ciaran, have fixed this in https://github.com/hashintel/hash/pull/3319/commits/3e51a34d3fe40ff57ffa9aea0120c2d95d957acf. Currently there is nothing to edit on the profile page except the avatar, which I've disabled for now because it depends on the modal existing. I've left a `@todo` and will address this in follow-up.